sql/plan/ddl.go

```diff
@@ -518,27 +529,34 @@ func (c *CreateTable) Temporary() TempTableOption {
 	return c.temporary
 }
 
-func (c *CreateTable) WithExpressions(exprs ...sql.Expression) (sql.Node, error) {
+func (c CreateTable) WithExpressions(exprs ...sql.Expression) (sql.Node, error) {
 	if len(exprs) != len(c.CreateSchema.Schema)+len(c.chDefs) {
 		return nil, sql.ErrInvalidChildrenNumber.New(c, len(exprs), len(c.CreateSchema.Schema)+len(c.chDefs))
 	}
 
-	nc := *c
+	nc := c
 
+	// Make sure to make a deep copy of any slices here so we aren't modifying the original pointer
+	ns := make(sql.Schema, len(c.CreateSchema.Schema))
```

Comment:
Seems like it would be more straightforward to have a `Schema.Copy()` or `Schema.Duplicate()` function.

Reply:
Good idea